src/main/scala/com/microsoft/hyperspace/package.scala

```diff
@@ -29,42 +30,53 @@ package object hyperspace {
   implicit class Implicits(sparkSession: SparkSession) {
 
     /**
-     * Plug in Hyperspace-specific rules.
+     * Plug in Hyperspace-specific rules and set `IndexConstants.HYPERSPACE_APPLY_ENABLED` as true.
      *
      * @return a spark session that contains Hyperspace-specific rules.
      */
     def enableHyperspace(): SparkSession = {
-      disableHyperspace
-      sparkSession.sessionState.experimentalMethods.extraOptimizations ++=
-        ApplyHyperspace :: Nil
-      sparkSession.sessionState.experimentalMethods.extraStrategies ++=
-        BucketUnionStrategy :: Nil
+      HyperspaceConf.setHyperspaceApplyEnabled(sparkSession, true)
+
+      if (!sparkSession.sessionState.experimentalMethods.extraOptimizations.contains(
+          ApplyHyperspace)) {
+        sparkSession.sessionState.experimentalMethods.extraOptimizations ++=
+          ApplyHyperspace :: Nil
+      }
+      if (!sparkSession.sessionState.experimentalMethods.extraStrategies.contains(
+          BucketUnionStrategy)) {
+        sparkSession.sessionState.experimentalMethods.extraStrategies ++=
+          BucketUnionStrategy :: Nil
+      }
       sparkSession
     }
 
     /**
-     * Plug out Hyperspace-specific rules.
+     * Set `IndexConstants.HYPERSPACE_APPLY_ENABLED` as false
+     * to stop ApplyHyperspace from applied to Spark plans.
+     * Note that `extraOptimizations` can contains ApplyHyperspace in it,
+     * even after Hyperspace is disabled.
```

Comment:
```suggestion
     * Disable Hyperspace indexes.
     *
     * Set `IndexConstants.HYPERSPACE_APPLY_ENABLED` as false
     * to stop applying Hyperspace indexes.
```
I don't think we should expose the impl detail in API comment. Maybe we can move the detailed comment inside the function.